### ISSUE TITLE
AUPRC `sample_weights` supported, but not used

### ISSUE THREAD

**Issue by cwognum** [Lightning-AI/metrics#1098]
## 🐛 Bug

The [AUPRC](https://torchmetrics.readthedocs.io/en/stable/classification/average_precision.html?highlight=average_precisionv) metric supports the `sample_weights` argument, but it is [never used](https://github.com/Lightning-AI/metrics/blob/master/src/torchmetrics/functional/classification/average_precision.py#L93). Using the `sample_weights` argument won't throw any error, nor does its documentation mention that its not yet actually supported. I think this is a silent bug that can cause confusion or invalidate results without people realizing it. 

#### Code sample
```python
import torch
from torchmetrics.functional import average_precision

pred = torch.tensor([1, 1, 0, 0])
target = torch.tensor([1, 0, 1, 0])
weights = torch.tensor([1, 0, 0, 0])

average_precision(pred, target)
# Returns: tensor(0.5000)

average_precision(pred, target, sample_weights=weights)
# Returns: tensor(0.5000)

```

Now if we compare this to [the implementation in Scikit-learn](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html?highlight=average_precision#sklearn.metrics.average_precision_score):
```python
import numpy as np
from sklearn.metrics import average_precision_score 

pred = np.array([1, 1, 0, 0])
target = np.array([1, 0, 1, 0])
weights = np.array([1, 0, 0, 0])

average_precision_score(pred, target)
# Returns: 0.5

average_precision_score(pred, target, sample_weights=weights)
# Returns: 1.0
```

### Expected behavior
I would expect a `NotImplementedError` to be thrown and the documentation to mention that it is not yet supported. 

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source): 
  - Version 0.9.1
  - Installed with `mamba`
  
- Python & PyTorch Version (e.g., 1.0): 
  - Python: 3.9.13
  - PyTorch: 1.11.0.post202
  
- Any other relevant information such as OS (e.g., Linux):
  - I am on Ubuntu, Linux.

**Comment by cwognum** [comment]
I would be happy to open a PR with my proposed fix once it is confirmed that this is indeed a bug.
